bot.go

```diff
@@ -294,212 +300,123 @@ func (bot *BotAPI) IsMessageToMe(message Message) bool {
 // Offset and Limit are optional.
 func (bot *BotAPI) GetUserProfilePhotos(config UserProfilePhotosConfig) (UserProfilePhotos, error) {
-	v := url.Values{}
-	v.Add("user_id", strconv.Itoa(config.UserID))
-	if config.Offset != 0 {
-		v.Add("offset", strconv.Itoa(config.Offset))
-	}
-	if config.Limit != 0 {
-		v.Add("limit", strconv.Itoa(config.Limit))
-	}
-
-	resp, err := bot.MakeRequest("getUserProfilePhotos", v)
+	resp, err := bot.Request(config)
 	if err != nil {
 		return UserProfilePhotos{}, err
 	}
 
 	var profilePhotos UserProfilePhotos
-	json.Unmarshal(resp.Result, &profilePhotos)
-
-	bot.debugLog("GetUserProfilePhoto", v, profilePhotos)
+	err = json.Unmarshal(resp.Result, &profilePhotos)
 
-	return profilePhotos, nil
+	return profilePhotos, err
 }
 
 // GetFile returns a File which can download a file from Telegram.
 //
 // Requires FileID.
 func (bot *BotAPI) GetFile(config FileConfig) (File, error) {
-	v := url.Values{}
-	v.Add("file_id", config.FileID)
-
-	resp, err := bot.MakeRequest("getFile", v)
+	resp, err := bot.Request(config)
 	if err != nil {
 		return File{}, err
 	}
 
 	var file File
-	json.Unmarshal(resp.Result, &file)
-
-	bot.debugLog("GetFile", v, file)
+	err = json.Unmarshal(resp.Result, &file)
 
-	return file, nil
+	return file, err
 }
 
 // GetUpdates fetches updates.
 // If a WebHook is set, this will not return any data!
 //
-// Offset, Limit, and Timeout are optional.
+// Offset, Limit, Timeout, and AllowedUpdates are optional.
 // To avoid stale items, set Offset to one higher than the previous item.
 // Set Timeout to a large number to reduce requests so you can get updates
 // instantly instead of having to wait between requests.
 func (bot *BotAPI) GetUpdates(config UpdateConfig) ([]Update, error) {
-	v := url.Values{}
-	if config.Offset != 0 {
-		v.Add("offset", strconv.Itoa(config.Offset))
-	}
-	if config.Limit > 0 {
-		v.Add("limit", strconv.Itoa(config.Limit))
-	}
-	if config.Timeout > 0 {
-		v.Add("timeout", strconv.Itoa(config.Timeout))
-	}
-
-	resp, err := bot.MakeRequest("getUpdates", v)
+	resp, err := bot.Request(config)
 	if err != nil {
 		return []Update{}, err
 	}
 
 	var updates []Update
-	json.Unmarshal(resp.Result, &updates)
-
-	bot.debugLog("getUpdates", v, updates)
-
-	return updates, nil
-}
+	err = json.Unmarshal(resp.Result, &updates)
 
-// RemoveWebhook unsets the webhook.
-func (bot *BotAPI) RemoveWebhook() (APIResponse, error) {
-	return bot.MakeRequest("deleteWebhook", url.Values{})
-}
-
-// SetWebhook sets a webhook.
-//
-// If this is set, GetUpdates will not get any data!
-//
-// If you do not have a legitimate TLS certificate, you need to include
-// your self signed certificate with the config.
-func (bot *BotAPI) SetWebhook(config WebhookConfig) (APIResponse, error) {
-
-	if config.Certificate == nil {
-		v := url.Values{}
-		v.Add("url", config.URL.String())
-		if config.MaxConnections != 0 {
-			v.Add("max_connections", strconv.Itoa(config.MaxConnections))
-		}
```

Comment:
The README example needs to be updated correspondingly:
https://github.com/go-telegram-bot-api/telegram-bot-api/blob/4fe428c77a68a9903ab5bfcaecb978003ce0424b/README.md#L90